packages/EasyApiToken/src/External/AwsCognitoJwtDriver.php

```diff
@@ -11,30 +11,33 @@
 use EonX\EasyApiToken\External\AwsCognito\JwkFetcher;
 use EonX\EasyApiToken\External\Interfaces\JwtDriverInterface;
 use Firebase\JWT\JWT;
+use Firebase\JWT\Key;
 
 final class AwsCognitoJwtDriver implements JwtDriverInterface
 {
+    use JwtTrait;
+
     private const TOKEN_TYPE_ACCESS = 'access';
 
     private const TOKEN_TYPE_ID = 'id';
 
     /**
      * @var string[]
+     *
+     * @deprecated will be removed in 5.0
      */
     private array $allowedAlgos;
 
-    private JwkFetcherInterface $jwkFetcher;
-
     /**
      * @param null|string[] $allowedAlgos
      */
     public function __construct(
-        private UserPoolConfigInterface $userPoolConfig,
-        ?JwkFetcherInterface $jwkFetcher = null,
+        private readonly UserPoolConfigInterface $userPoolConfig,
+        private JwkFetcherInterface $jwkFetcher = new JwkFetcher(),
         ?array $allowedAlgos = null,
-        private ?int $leeway = null
+        private readonly ?int $leeway = null,
+        private readonly string $defaultJwkAlgo = 'HS256'
```

Comment:
Shouldn't it be `RS256` by default for AwsCognitoJwtDriver? As I can see, we using `RS256` for Cognito JWT.
And, please, use the constant if possible

Reply:
Yes, you are right. Refactored.